### ISSUE TITLE
Rails5, sqlite3, ActiveRecord::ConnectionNotEstablished: No connection pool with id primary found

### ISSUE THREAD

**Issue by jrochkind** [jruby/activerecord-jdbc-adapter#747]
Oddly, I am getting a different error depending on whether I use a rails app skeleton or plugin skeleton in my reproduction. I've encountered both of these types of errors in my actual projects (some of which are gems/plugins), this was just an attempt to make a repro from as simple a rails app as possible. Seems out of the box `rails new appname` and `rails plugin new pluginname` are both sufficient to reproduce, in different ways. 
## 1. with app skeleton

``` bash
$ ruby -v
jruby 9.1.2.0 (2.3.0) 2016-05-26 7357c8f Java HotSpot(TM) 64-Bit Server VM 25.72-b15 on 1.8.0_72-b15 +jit [darwin-x86_64]

$ rails -v
Rails 5.0.0.1

$ rails new rails5_app_jruby_test
# [etc]

$ cd rails5_app_jruby_test
$ bundle show activerecord-jdbc-adapter
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21

$ rake test
DEPRECATION WARNING: alias_method_chain is deprecated. Please, use Module#prepend instead. From module, you can access the original method using super. (called from block in (root) at /Users/jrochkind/code/rails5jruby5k/config/application.rb:7)
rake aborted!
Bundler::GemRequireError: There was an error while trying to load the gem 'activerecord-jdbcsqlite3-adapter'.
Gem Load Error is: uninitialized constant ActiveRecord::ConnectionAdapters::Column::Format
Backtrace for gem load error is:
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc/type_cast.rb:13:in `<module:TypeCast>'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc/type_cast.rb:7:in `<module:Jdbc>'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc/type_cast.rb:4:in `<module:ConnectionAdapters>'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc/type_cast.rb:3:in `<top>'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc/column.rb:1:in `singleton class'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc/column.rb:70:in `<class:singleton class>'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc/column.rb:68:in `<module:JdbcColumn>'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc/column.rb:12:in `<module:ConnectionAdapters>'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc/column.rb:2:in `<top>'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc/column.rb:1:in `<top>'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc/adapter.rb:1:in `require'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc/adapter.rb:1:in `<module:(root)>'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc/adapter.rb:9:in `<top>'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc.rb:1:in `require'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc.rb:1:in `<top>'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc.rb:52:in `require'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc.rb:52:in `<top>'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc.rb:3:in `require'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc/jdbc.rb:3:in `block in (root)'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc.rb:1:in `each'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc.rb:1:in `block in (root)'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc.rb:9:in `each'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21/lib/arjdbc.rb:9:in `(root)'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbcsqlite3-adapter-1.3.21/lib/activerecord-jdbcsqlite3-adapter.rb:1:in `(root)'
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbcsqlite3-adapter-1.3.21/lib/activerecord-jdbcsqlite3-adapter.rb:2:in `<top>'

# [...]
```
## 2. From a `rails plugin new` skeleton

``` bash
$ ruby -v
jruby 9.1.2.0 (2.3.0) 2016-05-26 7357c8f Java HotSpot(TM) 64-Bit Server VM 25.72-b15 on 1.8.0_72-b15 +jit [darwin-x86_64]

$ rails -v
Rails 5.0.0.1

$ rails plugin new rails5jrubyplugin
# [etc]

$ cd rails5jrubyplugin
$ bundle show activerecord-jdbc-adapter
/Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-jdbc-adapter-1.3.21

$ rake test
/Users/jrochkind/.gem/jruby/2.3.0/gems/activesupport-5.0.0.1/lib/active_support/core_ext/date_and_time/calculations.rb:18: warning: Ambiguous first argument; make sure.
/Users/jrochkind/.gem/jruby/2.3.0/gems/activesupport-5.0.0.1/lib/active_support/core_ext/date_and_time/calculations.rb:23: warning: Ambiguous first argument; make sure.
DEPRECATION WARNING: alias_method_chain is deprecated. Please, use Module#prepend instead. From module, you can access the original method using super. (called from alias_method_chain at /Users/jrochkind/.gem/jruby/2.3.0/gems/activesupport-5.0.0.1/lib/active_support/core_ext/module/aliasing.rb:27)
ActiveRecord::ConnectionNotEstablished: No connection pool with id primary found.
                                                                       retrieve_connection at /Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-5.0.0.1/lib/active_record/connection_adapters/abstract/connection_pool.rb:874
                                                                       retrieve_connection at /Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-5.0.0.1/lib/active_record/connection_handling.rb:128
                                                                                connection at /Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-5.0.0.1/lib/active_record/connection_handling.rb:91
                                                                          needs_migration? at /Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-5.0.0.1/lib/active_record/migration.rb:576
                                                                   load_schema_if_pending! at /Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-5.0.0.1/lib/active_record/migration.rb:576
                                                            block in maintain_test_schema! at /Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-5.0.0.1/lib/active_record/migration.rb:591
                                                                         suppress_messages at /Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-5.0.0.1/lib/active_record/migration.rb:822
                                                                            method_missing at /Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-5.0.0.1/lib/active_record/migration.rb:596
                                                                     maintain_test_schema! at /Users/jrochkind/.gem/jruby/2.3.0/gems/activerecord-5.0.0.1/lib/active_record/migration.rb:591
                                                                                     <top> at /Users/jrochkind/.gem/jruby/2.3.0/gems/railties-5.0.0.1/lib/rails/test_help.rb:15
                                                                                   require at org/jruby/RubyKernel.java:944
                                                                          block in require at /Users/jrochkind/.gem/jruby/2.3.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:293
                                                                           load_dependency at /Users/jrochkind/.gem/jruby/2.3.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:259
                                                                                   require at /Users/jrochkind/.gem/jruby/2.3.0/gems/activesupport-5.0.0.1/lib/active_support/dependencies.rb:293
                                                                                     <top> at /Users/jrochkind/code/rails5jrubyplugin/test/test_helper.rb:1
                                                                                   require at org/jruby/RubyKernel.java:944
                                                                                     <top> at /Users/jrochkind/code/rails5jrubyplugin/test/test_helper.rb:6
                                                                                   require at org/jruby/RubyKernel.java:944
  block in /Users/jrochkind/.gem/jruby/2.3.0/gems/rake-11.2.2/lib/rake/rake_test_loader.rb at /Users/jrochkind/.gem/jruby/2.3.0/gems/rake-11.2.2/lib/rake/rake_test_loader.rb:10
                                                                                      each at org/jruby/RubyArray.java:1593
  block in /Users/jrochkind/.gem/jruby/2.3.0/gems/rake-11.2.2/lib/rake/rake_test_loader.rb at /Users/jrochkind/.gem/jruby/2.3.0/gems/rake-11.2.2/lib/rake/rake_test_loader.rb:9
                                                                                    select at org/jruby/RubyArray.java:2411
                                                                                     <top> at /Users/jrochkind/.gem/jruby/2.3.0/gems/rake-11.2.2/lib/rake/rake_test_loader.rb:4
rake aborted!
Command failed with status (1): [ruby -w -I"lib:lib:test" -I"/Users/jrochkind/.gem/jruby/2.3.0/gems/rake-11.2.2/lib" "/Users/jrochkind/.gem/jruby/2.3.0/gems/rake-11.2.2/lib/rake/rake_test_loader.rb" "test/**/*_test.rb" ]
/Users/jrochkind/.gem/jruby/2.3.0/gems/rake-11.2.2/exe/rake:27:in `<top>'
Tasks: TOP => test
(See full trace by running task with --trace)
```
##

**Comment by enebo** [comment]
@jrochkind our current release of arjdbc is known to not work with rails5.  We have a rails5 branch (which seems to handle common cases [except views] on sqlite3) where hopefully soonish we will get much more serious in finishing work and getting a release out.  We are behind in regards to rails5 on arjdbc so your problem is known.

**Comment by enebo** [comment]
Speculatively closing since I am doing pretty much everything correctly in the soon to be released 50.0 release.  Please re-open if that does not fix this issue.
